# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/rustc/no-bashcompletion.patch
# Copyright (C) 2021 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- rustc-1.75.0-src/src/bootstrap/src/utils/tarball.rs.vanilla	2024-02-07 16:52:41.132878313 +0100
+++ rustc-1.75.0-src/src/bootstrap/src/utils/tarball.rs	2024-02-07 16:54:06.136882698 +0100
@@ -178,7 +178,7 @@
         t!(std::fs::create_dir_all(&destdir));
         self.builder.install(src.as_ref(), &destdir, perms);
     }
-
+/*
     pub(crate) fn add_renamed_file(
         &self,
         src: impl AsRef<Path>,
@@ -189,7 +189,7 @@
         t!(std::fs::create_dir_all(&destdir));
         self.builder.copy(src.as_ref(), &destdir.join(new_name));
     }
-
+*/
     pub(crate) fn add_legal_and_readme_to(&self, destdir: impl AsRef<Path>) {
         for file in self.overlay.legal_and_readme() {
             self.add_file(self.builder.src.join(file), destdir.as_ref(), 0o644);

--- rustc-1.78.0-src/src/bootstrap/src/core/build_steps/dist.rs.vanilla	2024-05-18 18:45:21.948931045 +0200
+++ rustc-1.78.0-src/src/bootstrap/src/core/build_steps/dist.rs	2024-05-18 18:45:29.919930643 +0200
@@ -1071,8 +1071,8 @@
         tarball.set_overlay(OverlayKind::Cargo);
 
         tarball.add_file(cargo, "bin", 0o755);
-        tarball.add_file(etc.join("_cargo"), "share/zsh/site-functions", 0o644);
-        tarball.add_renamed_file(etc.join("cargo.bashcomp.sh"), "etc/bash_completion.d", "cargo");
+        //tarball.add_file(etc.join("_cargo"), "share/zsh/site-functions", 0o644);
+        //tarball.add_renamed_file(etc.join("cargo.bashcomp.sh"), "etc/bash_completion.d", "cargo");
         tarball.add_dir(etc.join("man"), "share/man/man1");
         tarball.add_legal_and_readme_to("share/doc/cargo");
 
